app/index.js

```diff
@@ -1,12 +1,27 @@
-import React from 'react'
-import { render } from 'react-dom'
-import './styles'
-import Header from './components/Header/Header'
-
-const App = () => {
-  return (
-    <Header />
-  );
-}
-
-render(<App />, document.querySelector('.application'));
+import React from 'react';
+import { Router, Route, IndexRoute, browserHistory } from 'react-router';
+import { Provider } from 'react-redux';
+import { createStore } from 'redux';
+import rootReducer from './reducers';
+import ReactDOM, { render } from 'react-dom';
+
+import Header from './components/Header/Header';
+import App from './components/app';
+import Login from './components/login';
+import './styles';
+
+const devTools = window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__();
+
+const store = createStore(rootReducer, devTools);
+
+const router = (
+  <Provider store={store}>
+    <Router history={browserHistory} >
+      <Route path='/' component={App} >
```

Comment:
Your problem is that Login is a child component to App. When routed to '/login' it needs to know where to place itself inside the App component.